plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/LibraryFile.java

```diff
@@ -0,0 +1,67 @@
+package com.google.cloud.tools.eclipse.appengine.libraries;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * Represent a jar file that is downloaded from the location defined by {@link MavenCoordinates}. It can have associated
+ * filters to control visibility of classes and packages contained in the jar file.
+ *
+ */
+public class LibraryFile {
+
+  private List<Filter> exclusionFilters = Collections.emptyList();
+  private List<Filter> inclusionFilters = Collections.emptyList();
```

Comment:
As mentioned above, these filters need to be woven together and their order matters.